HealthcareSystem/Backend/Service/ActionBenefitService/RabbitMqActionBenefitService/RabbitMqActionBenefitService.cs

```diff
@@ -0,0 +1,138 @@
+using RabbitMQ.Client;
+using RabbitMQ.Client.Events;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Backend.Service
+{
+    public class RabbitMqActionBenefitMessageingService : BackgroundService
+    {
+        private IModel _channel;
+        private IConnection _connection;
+        private readonly IServiceProvider _service;
+        private readonly String _hostname;
+        private readonly String _username;
+        private readonly String _password;
+        private readonly String _queueName;
+
+        public RabbitMqActionBenefitMessageingService(IServiceProvider service, IOptions<RabbitMqConfiguration> rabbitMqOPtions)
+        {
+            _service = service;
+            _hostname = rabbitMqOPtions.Value.Hostname;
+            _username = rabbitMqOPtions.Value.Username;
+            _password = rabbitMqOPtions.Value.Password;
+            _queueName = "bolnica-1";
+            InitializeRabbitMqListener();
+        }
+
+        private void InitializeRabbitMqListener()
+        {
+            var factory = new ConnectionFactory() { HostName = _hostname, UserName = _username, Password = _password };
+            _connection = factory.CreateConnection();
+            _connection.ConnectionShutdown += RabbitMQ_ConnectionShutdown;
+            _channel = _connection.CreateModel();
+            _channel.QueueDeclare(queue: _queueName,
+                                  durable: true,
+                                  exclusive: false,
+                                  autoDelete: false,
+                                  arguments: null);
+
+            using (var scope = _service.CreateScope())
+            {
+                IPharmacyService pharmacyService = scope.ServiceProvider.GetRequiredService<IPharmacyService>();
+
+                foreach (Pharmacy p in pharmacyService.GetPharmaciesBySubscribed(true))
+                {
+                    _channel.ExchangeDeclare(exchange: p.ActionsBenefitsExchangeName, type: ExchangeType.Fanout);
+                    _channel.QueueBind(queue: _queueName, exchange: p.ActionsBenefitsExchangeName, routingKey: "");
+                }
+            }
+        }
+
+        protected override Task ExecuteAsync(CancellationToken stoppingToken)
+        {
+            stoppingToken.ThrowIfCancellationRequested();
+
+            var consumer = new EventingBasicConsumer(_channel);
+            consumer.Received += OnConsumerReceived;
+            consumer.Shutdown += OnConsumerShutdown;
+            consumer.Registered += OnConsumerRegistered;
+            consumer.Unregistered += OnConsumerUnregistered;
+            consumer.ConsumerCancelled += OnConsumerCancelled;
+
+            _channel.BasicConsume(queue: _queueName, autoAck: false, consumer: consumer);
+
+            return Task.CompletedTask;
+        }
+
+        public void Unsubscribe(String exchangeName)

```

Comment:
Should there be a Subscribe method as well? I feel like that is probably a use case as well, like when a new Pharmacy is added and subscribed to during runtime?

Reply:
yes it will be later, but that is part of different feature, so ill add it when i do that feature